rfcs/0121-hardware-driver.md

```diff
@@ -0,0 +1,86 @@
+---
+feature: Use a more canonical name for hardware driver path and modules
+start-date: 2022-02-03
+author: @jonringer
+co-authors:
+shepherd-team:
+shepherd-leader:
+related-issues: https://github.com/NixOS/nixpkgs/issues/141803
+---
+
+# Summary
+[summary]: #summary
+
+Currently, NixOS mounts video drivers under the path `/run/opengl-driver/`,
+but should be mounted under a more generic '/run/hardware-drivers/` path.
+The usage of opengl explicitly may have been an accurate name given the time
+in which NixOS was first created; however, graphic drivers alone include much
+more than just userland graphics libraries so this usages is now misaligned with
+current paradigms. The misalignment with this path and the related Nixpkgs
+utilities are a misnomer in most contexts, and contributes to Nixpkgs' 
+"weirdness budget".
+
+# Motivation
+[motivation]: #motivation
+
+Video drivers ship many hardware acceleration libraries these days: vulkan,
+opengl, opencl, video encoding, cuda, and many other userland libraries are
+installed. Having all of these libraries placed under a single "opengl" header
+is odd for many contributors, especially newcomers. This awkwardness is
+compounded with other specialized hardware such as fpgas, tpus, and asics
+where their function isn't related to graphics, however, also need to make
+use of a "known good" path in which the related userland libraries will be mounted.
+
+To remedy this misalignment, a new convention around placing those libraries
+should be used, preferably `/run/hardware-drivers/`.
+
+# Detailed design
+[design]: #detailed-design
+
+Most of the implementation of the current paradigms will continue, as there
+will always be a need for some "impurity" around hardware based libraries.
+The significant changes will be around naming conventions used within
+Nixpkgs and NixOS.
+
+Deprecate and move existing `/run/opengl-drivers/` logic:
+- Rename `hardware.opengl` options to `hardware.drivers`
```

Comment:
Maybe instead it should be `hardware.gpu.drivers` or something? We already have `hardware.cpu` for e.g. microcode options. Would be nice if we could make it somewhat symmetrical.

Reply:
Yea, I was thinking of something similar

Reply:
really jarring that for sway, you need to set something under xserver

Reply:
Added a comment, I'll reflect it in the PR I'm building up